### PR TITLE
Add templates - GitHub issues and PRs

### PR DESCRIPTION
**Pull request by jlucktay**
## Description

Added some templates for [new defect and feature issues, and PRs](https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/about-issue-and-pull-request-templates).

## Related Issue

N/A

## Motivation and Context

Having checklists and section headers to prompt contributors with (rather than just a blank slate) will aid information gathering and feedback.

## Test Coverage For This Change

N/A

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [x] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [x] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor` in applicable directories.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [x] `gofmt -s -w .`
  - [x] `go vet ./...`
  - [ ] `golangci-lint run`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Changelog
```
Comment (asoorm) [inline CHANGELOG.md:1]: Thanks - this changelog will need manual maintenance right? At the moment, every release, we generate the changelog with the release notes.

Screenshot 2020-12-18 at 17 49 02

Comment (asoorm) [reply CHANGELOG.md:1]: https://github.com/TykTechnologies/tyk-operator/releases/tag/v0.4.0 as an example.

Is there a benefit in doing it one way over the other?

File: .github/ISSUE_TEMPLATE/FEATURE.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+---
+name: 🚀 Feature request
```
Comment (asoorm) [inline .github/ISSUE_TEMPLATE/FEATURE.md:2]: I think maybe larger feature requests can go into https://github.com/TykTechnologies/tyk-operator/discussions

If it's a small feature request, We should enhance this template to encourage feature requests to follow the BDD format.

sample:
```
Feature: Managing http APIs
  In order to control the Tyk Gateway
  As a developer
  I need to be able to create, update and delete API resources
  And the gateway should reconcile accordingly
  Scenario: Create a keyless api
    Given there is a ./custom_resources/httpbin.keyless.apidefinition.yaml resource
    When i request /httpbin/get endpoint
    Then there should be a 200 http response code

  Scenario: Update an api from keyless to auth token
    Given there is a ./custom_resources/httpbin.keyless.apidefinition.yaml resource
    When i update a ./custom_resources/httpbin.apikey.apidefinition.yaml resource
      And i request /httpbin/get endpoint
    Then there should be a 401 http response code

  Scenario: Delete an API
    Given there is a ./custom_resources/httpbin.keyless.apidefinition.yaml resource
    When i delete a ./custom_resources/httpbin.keyless.apidefinition.yaml resource
      And i request /httpbin/get endpoint
    Then there should be a 404 http response code

  Scenario: Transform xml to json
    Given there is a ./../config/samples/httpbin_transform.yaml resource
    When i request /httpbin-transform/xml endpoint
    Then there should be a "Content-Type: application/json" response header
```

File: CHANGELOG.md (hunk 1 above)
Comment (jlucktay) [reply CHANGELOG.md:1]: If you're happy with the subject lines from the commits/PRs forming the body of the changelog like this, then you don't need to re-invent this particular wheel 🙂 
The [0.4.0](https://github.com/TykTechnologies/tyk-operator/releases/tag/v0.4.0) and [0.4.1](https://github.com/TykTechnologies/tyk-operator/releases/tag/v0.4.1) releases do look like they have plenty of meat on the bone as-is 👍

File: .github/ISSUE_TEMPLATE/FEATURE.md (hunk 2 above)
Comment (jlucktay) [reply .github/ISSUE_TEMPLATE/FEATURE.md:2]: RFCs via the discussions feature - yep, love it 💯  will add a note and a link to direct larger changes in that direction.
Will add a slimmed-down BDD skeleton to the template as well.

File: CHANGELOG.md (hunk 1 above)
Comment (jlucktay) [reply CHANGELOG.md]: 9bbdba5 resolves

File: .github/ISSUE_TEMPLATE/FEATURE.md (hunk 2 above)
Comment (jlucktay) [reply .github/ISSUE_TEMPLATE/FEATURE.md:2]: f78058b and 063c5cd should resolve.

**Comment by asoorm** [comment]
LGTM - Thanks @jlucktay 

@gernest can you review and check you are happy with these templates before we merge it?

**Review by gernest — APPROVED** [review]
LGTM

**Review by asoorm — APPROVED** [review]

**Pull request merged**